flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -149,6 +156,12 @@ public void initializeState(StateInitializationContext context) throws Exception
     // Open the table loader and load the table.
     this.tableLoader.open();
     this.table = tableLoader.loadTable();
+    if (reloadIntervalMs > 0) {
+      this.tableSupplier = new ReloadingTableSupplier(table, tableLoader, reloadIntervalMs);
```

Comment:
Why do not get the `tableSupplier` from outside? Why do we create it here and in the `FlinkSink` as well?

QQ: What is the difference between the functionality of the `TableLoader` and the `TableSupplier`? Is it just a TableLoader with cache?

Reply:
`TableLoader` is specifically for loading a table from a catalog. A `TableSupplier` implementation might get the table from someplace else, like the delegation framework or a broadcast (the initial TableSupplier implementation uses a TableLoader).

Reply:
The first part of the comment is relevant anyway:
- We should not create the supplier in the committer - we should get it from the Sink as an initialization parameter. We should keep the supplier creation in one place.

About the Loader vs. Supplier, I think from the consumer side they have the same functionality - get the table from the source of truth.... Should we use the same interface?

Reply:
I removed the table supplier in the committer, it only uses a table loader now. In the writer, the supplier is owned by the task writer factory, not the operator, but the operator is responsible for triggering the refresh. The current TableLoader interface doesn't provide a good way to do that.